implementations/micrometer-registry-dynatrace/src/main/java/io/micrometer/dynatrace/types/DynatraceTimer.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2022 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.dynatrace.types;
+
+import io.micrometer.core.instrument.AbstractMeter;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+/**
+ * Resettable {@link Timer} implementation for Dynatrace exporters.
+ *
+ * @author Georg Pirklbauer
+ * @since 1.9.0
+ */
+public final class DynatraceTimer extends AbstractMeter implements Timer, DynatraceSummarySnapshotSupport {
+    private final DynatraceSummary summary = new DynatraceSummary();
+    private final Clock clock;
+    private final TimeUnit baseTimeUnit;
+
+    public DynatraceTimer(Id id, Clock clock, TimeUnit baseTimeUnit) {
+        super(id);
+        this.clock = clock;
+        this.baseTimeUnit = baseTimeUnit;
+    }
+
+    @Override
+    public boolean hasValues() {
+        return count() > 0;
+    }
+
+    @Override
+    public DynatraceSummarySnapshot takeSummarySnapshot() {
+        return takeSummarySnapshot(baseTimeUnit());
+    }
+
+    @Override
+    public DynatraceSummarySnapshot takeSummarySnapshot(TimeUnit unit) {
+        return new DynatraceSummarySnapshot(min(unit), max(unit), totalTime(unit), count());
+    }
+
+    @Override
+    public DynatraceSummarySnapshot takeSummarySnapshotAndReset() {
+        return takeSummarySnapshotAndReset(baseTimeUnit());
+    }
+
+    @Override
+    public DynatraceSummarySnapshot takeSummarySnapshotAndReset(TimeUnit unit) {
+        DynatraceSummarySnapshot snapshot = takeSummarySnapshot(unit);
+        summary.reset();
+        return snapshot;
+    }
+
+    // from AbstractTimer
+    @Override
+    public <T> T recordCallable(Callable<T> f) throws Exception {
```

Comment:
This is exactly the same code as the `recordCallable` in `AbstractTimer`, you don't need to override it (I guess the same is true for the other record methods).